plugins/generator-1.18.2/forge-1.18.2/templates/elementinits/villagerprofessions.java.ftl

```diff
@@ -0,0 +1,84 @@
+<#--
+ # MCreator (https://mcreator.net/)
+ # Copyright (C) 2012-2020, Pylo
+ # Copyright (C) 2020-2022, Pylo, opensource contributors
+ #
+ # This program is free software: you can redistribute it and/or modify
+ # it under the terms of the GNU General Public License as published by
+ # the Free Software Foundation, either version 3 of the License, or
+ # (at your option) any later version.
+ #
+ # This program is distributed in the hope that it will be useful,
+ # but WITHOUT ANY WARRANTY; without even the implied warranty of
+ # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ # GNU General Public License for more details.
+ #
+ # You should have received a copy of the GNU General Public License
+ # along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ #
+ # Additional permission for code generator templates (*.ftl files)
+ #
+ # As a special exception, you may create a larger work that contains part or
+ # all of the MCreator code generator templates (*.ftl files) and distribute
+ # that work under terms of your choice, so long as that work isn't itself a
+ # template for code generation. Alternatively, if you modify or redistribute
+ # the template itself, you may (at your option) remove this special exception,
+ # which will cause the template and the resulting code generator output files
+ # to be licensed under the GNU General Public License without this special
+ # exception.
+-->
+
+<#-- @formatter:off -->
+
+<#include "../mcitems.ftl">
+
+/*
+*    MCreator note: This file will be REGENERATED on each build.
+*/
+
+package ${package}.init;
+
+import net.minecraft.sounds.SoundEvent;
+
+public class ${JavaModName}VillagerProfessions {
+
+    public static final DeferredRegister<PoiType> POI = DeferredRegister.create(ForgeRegistries.POI_TYPES, ${JavaModName}.MODID);
+    public static final DeferredRegister<VillagerProfession> PROFESSIONS = DeferredRegister.create(ForgeRegistries.PROFESSIONS, ${JavaModName}.MODID);
+
+    <#list villagerprofessions as villagerprofession>
+        public static final RegistryObject<VillagerProfession> ${villagerprofession.getModElement().getRegistryNameUpper()} =
+            registerProfession(
+			    "${villagerprofession.getModElement().getRegistryName()}",
+                ${mappedBlockToBlock(villagerprofession.pointOfInterest)},
+                () -> new SoundEvent(new ResourceLocation("${villagerprofession.actionSound}"))
+            );
+    </#list>
+
+	private static RegistryObject<VillagerProfession> registerProfession(String name, Block block, Supplier<SoundEvent> soundEventSupplier) {
+		Optional<PoiType> existingCheck = PoiType.forState(block.defaultBlockState());
+
+		if (existingCheck.isPresent()) {
+            ${JavaModName}.LOGGER.error("Skipping villager profession " + name + " that uses POI block " + block + " that is already in use by " + existingCheck);
```

Comment:
I think I'd warn here instead as we don't handle any exception objects here.

Reply:
It is error as profession will be missing in the game